### DISCUSSION TITLE
Using KerasTuner for Batchdataset in Tensorflow 2.0

### DISCUSSION THREAD

**Discussion by live2awesome** [keras-team/keras-tuner#752] [Q&A] [answered]
Hi Guys I am trying to add batch dataset combine kerastuner bayesianoptimization and tensorflow ?

X is 
<BatchDataset shapes: ({Sales: (None,), Quantity: (None,), Discount: (None,), Profit: (None,)}, (None,)), types: ({Sales: tf.float64, Quantity: tf.int64, Discount: tf.float64, Profit: tf.float64}, tf.int32)>

Y is 
<BatchDataset shapes: (None,), types: tf.int32> Tensors are batch size is 32 . While passing into tuner.search() I am receiving error 

`y` argument is not supported when using dataset as input.

Value of Y :
tf.Tensor([1 3 3 3 3 3 3 3 3 1 1 3 3 3 3 3 2 2 3 3 1 3 3 3 3 0 1 3 1 0 3 1], shape=(32,), dtype=int32)
tf.Tensor([3 1 3 3 3 3 3 1 1 3 3 0 3 1 0 3 3 1 0 2 1 3 3 1 1 0 3 3 3 3 3 1], shape=(32,), dtype=int32)
tf.Tensor([3 3 0 1 3 3 3 1 3 0 3 3 3 1 3 3 3 3 0 3 1 3 3 0 3 3 3 3 1 3 0 3], shape=(32,), dtype=int32)
tf.Tensor([1 3 1 0 3 3 3 3 3 1 3 3 3 1 3 3 1 3 3 3 3 3 3 3 3 3 3 0 1 3 3 3], shape=(32,), dtype=int32)
tf.Tensor([1 3 1 3 3 3 1 3 1 1 3 1 1 1 3 3 1 3 0 3 3 1 1 3 0 1 3 3 1 3 3 3], shape=(32,), dtype=int32)

**Comment by haifeng-jin** [comment, marked as answer]
X and Y will be directly passed to the `model.fit()` function of a Keras model.
Please make sure it works without KerasTuner.
Something like:

```py
model = build_model(keras_tuner.HyperParameters())
model.compile(...)
model.fit(x, y, ...)
```
